SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPCTypes.h

```diff
@@ -9,29 +9,39 @@
 
 namespace SpatialGDK
 {
+struct ReceivedRPC
+{
+	uint32 Offset;
+	uint32 Index;
+	TArrayView<const uint8> PayloadData;
+};
+
 namespace RPCCallbacks
 {
 using DataWritten = TFunction<void(Worker_EntityId, Worker_ComponentId, Schema_ComponentData*)>;
 using UpdateWritten = TFunction<void(Worker_EntityId, Worker_ComponentId, Schema_ComponentUpdate*)>;
 using RequestWritten = TFunction<void(Worker_EntityId, Schema_CommandRequest*)>;
 using ResponseWritten = TFunction<void(Worker_EntityId, Schema_CommandResponse*)>;
-using RPCWritten = TFunction<void(Worker_EntityId, Worker_ComponentId, uint32)>;
+using RPCWritten = TFunction<void(Worker_ComponentId, uint64)>;
 
 using CanExtractRPCs = TFunction<bool(Worker_EntityId)>;
-using ProcessRPC = TFunction<bool(Worker_EntityId, const RPCPayload&)>;
 } // namespace RPCCallbacks
+
 /**
  * Structure encapsulating a read operation
  */
 struct RPCReadingContext
 {
+	FName ReaderName;
```

Comment:
This just used for debugability?

Reply:
I want to use it to replace RPCTypes, it could also be used for generating IDs for event tracing.